### PR TITLE
sensors/test_status: get the late status

### LINKED ISSUE #1: sensors/test_status: QA Notice leads to a false fail detection
```
[044] replication/box_set_replication_stress.test.lua memtx           [ pass ]
[044] replication/box_set_replication_stress.test.lua vinyl           
[044] * [QA Notice]
[044] *
[044] * The server 'master_quorum1' does not stop during 5 seconds after the 15 (SIGTERM) signal.
[044] * Info: process 36076 [unknown; unknown]
[044] * Sending SIGKILL...
[044] *
[044] [ pass ]
```

https://github.com/tarantool/tarantool/runs/3814050318

When there is no particular status, we don't search it somewhere below by the log, but assume fail:

https://github.com/tarantool/multivac/blob/06a1e7e5a6a95246f6e60a3a25d79679ae200168/multivac/sensors/test_status.py#L15

https://github.com/tarantool/multivac/blob/06a1e7e5a6a95246f6e60a3a25d79679ae200168/multivac/sensors/test_status.py#L64

Because usually such line break means that test-run print some information about a fail.

Given QA notice may actually be result of some problem, but we have a [notice](https://github.com/tarantool/test-run/blob/c34500365efe8316e79c7936a2f2d04644602936/lib/sampler.py#L58-L62) about low sampling resolution, which definitely should not be detected as a test fail.

In fact, there is no much sense to print a line in two chunks in test-run (see [here](https://github.com/tarantool/test-run/blob/c34500365efe8316e79c7936a2f2d04644602936/lib/test_suite.py#L256) and [here](https://github.com/tarantool/test-run/blob/c34500365efe8316e79c7936a2f2d04644602936/lib/test.py#L245)), because now the output is anyway line buffered (due to parallel running of tests). So maybe it worth to resolve it from the test-run side.

### PR DESCRIPTION
**Pull request by lastoCHka42**
Sometimes we get the test result few lines later, i.e. when there are QA Notice before the test result. This patch saves all the tests without the result to the dictionary and finds the result without test with the same worker ID.

Resolves #1

### CODE REVIEW

File: multivac/sensors/test_status.py [hunk 1]
```diff
@@ -58,15 +61,27 @@ def test_status_iter(log_fh, cache_filepath=None):
         cache = []
 
     hang_detected = False
+
+    saved_data = {}
```
Comment (ochaplashkin) [inline multivac/sensors/test_status.py:65]: `saved_data` is name of a variable that does not define its contents.
It is also not a good practice to have `data/dict/list` in the var name.

File: multivac/sensors/test_status.py [hunk 2]
```diff
@@ -58,15 +61,27 @@ def test_status_iter(log_fh, cache_filepath=None):
...
+            status = m.group('status')
             res = (m['test'], m['conf'], status)
             if cache_filepath:
                 cache.append(res)
             yield res
             continue
+        elif saved_data:
+            status_match = LATE_STATUS.match(line)
+            if status_match:
+                get_test = saved_data.pop(status_match.group('wid'))
```
Comment (ochaplashkin) [inline multivac/sensors/test_status.py:82]: `get_test` ([verb_..]) - good name for the function/method not for the variable

File: multivac/sensors/test_status.py [hunk 3]
```diff
@@ -91,6 +106,11 @@ def test_status_iter(log_fh, cache_filepath=None):
                 yield res
                 continue
 
+    # if there are tests with no result, save them as failed.
+    for wid in saved_data:
+        res = saved_data.get(wid)  # get tuple (test name, test conf)
+        yield (res[0], res[1], 'failed')
```
Comment (ochaplashkin) [inline multivac/sensors/test_status.py:112]: 1. Are you sure that it is `failed` and not `fail`? Test run print `[ fail ]` about test;
2. `failed` - magic string. Can you put this into any constant? I'm not sure how to do it better.

File: multivac/sensors/test_status.py (hunk 1 above)
Comment (lastoCHka42) [reply multivac/sensors/test_status.py]: fixed

File: multivac/sensors/test_status.py (hunk 2 above)
Comment (lastoCHka42) [reply multivac/sensors/test_status.py]: fixed

File: multivac/sensors/test_status.py (hunk 3 above)
Comment (lastoCHka42) [reply multivac/sensors/test_status.py]: fixed

**Review by ochaplashkin — APPROVED** [review]
LGTM :rocket:

**Pull request merged**